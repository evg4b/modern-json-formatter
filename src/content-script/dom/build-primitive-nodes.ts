import { createElement } from '@core/dom';
import { trim } from '@core/helpers';
import { type BooleanNode, type NumberNode, type StringNode } from '@worker-core';

export const buildNullNode = () =>
  createElement({
    element: 'span',
    content: 'null',
    class: 'null',
  });

export const buildNumberNode = ({ value }: NumberNode) =>
  createElement({
    element: 'span',
    content: value,
    class: 'number',
  });

export const buildBoolNode = (object: BooleanNode) =>
  createElement({
    element: 'span',
    content: object.value ? 'true' : 'false',
    class: 'boolean',
  });

export const buildStringNode = ({ value, variant }: StringNode) => {
  if (variant === 'url') {
    return createElement({
      element: 'span',
      content: JSON.stringify(value),
      class: ['string', 'url'],
      attributes: {
        href: trim(value, ' \t\n\r'),
      },
    });
  }

  if (variant === 'email') {
    return createElement({
      element: 'span',
      content: JSON.stringify(value),
      class: ['string', 'email'],
      attributes: {
        href: `mailto:${value}`,
      },
    });
  }

  return createElement({
    element: 'span',
    content: JSON.stringify(value),
    class: 'string',
  });
};
