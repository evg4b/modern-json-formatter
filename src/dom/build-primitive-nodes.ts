import { BooleanNode, NumberNode, StringNode } from '@packages/tokenizer';
import { element } from './helpres';

export const buildNullNode = () => element({ content: 'null', class: 'null' });

export const buildNumberNode = ({ value }: NumberNode) => element({ content: value, class: 'number' });

export const buildBoolNode = (object: BooleanNode) =>
  element({
    content: object.value ? 'true' : 'false',
    class: 'boolean',
  });

export const buildStringNode = ({ value }: StringNode) =>
  element({
    content: JSON.stringify(value),
    class: 'string',
  });
