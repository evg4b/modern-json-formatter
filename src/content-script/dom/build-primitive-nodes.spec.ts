import { describe, expect, it } from '@jest/globals';
import { buildStringNode } from './build-primitive-nodes';

describe('buildStringNode', () => {
  it('should create a URL string node', () => {
    const value = ' https://example.com ';
    const variant = 'url';

    const node = buildStringNode({ type: 'string', value, variant });

    expect(node).toMatchSnapshot();
  });

  it('should create an email string node', () => {
    const value = 'test@example.com';
    const variant = 'email';

    const node = buildStringNode({ type: 'string', value, variant });

    expect(node).toMatchSnapshot();
  });

  it('should create a default string node', () => {
    const value = 'default string';

    const node = buildStringNode({ type: 'string', value });

    expect(node).toMatchSnapshot();
  });
});
