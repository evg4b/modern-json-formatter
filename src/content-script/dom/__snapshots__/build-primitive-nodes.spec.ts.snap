// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildStringNode should create a URL string node 1`] = `
<span
  class="string url"
  href="https://example.com"
>
  " https://example.com "
</span>
`;

exports[`buildStringNode should create a default string node 1`] = `
<span
  class="string"
>
  "default string"
</span>
`;

exports[`buildStringNode should create an email string node 1`] = `
<span
  class="string email"
  href="mailto:test@example.com"
>
  "test@example.com"
</span>
`;
